### PR TITLE
A (tiny) effort to reduce memory usage per flow

### PR DESCRIPTION
**Pull request by IvanNardi**
Follow-up of 22241a1d
Only trivial changes:
* remove completely unused fields
* remove fields only written (but never read)
* CSGO protocol only handles UDP traffic

### CODE REVIEW

File: src/lib/protocols/rtsp.c [hunk 1]
```diff
@@ -84,20 +81,7 @@ void ndpi_search_rtsp_tcp_udp(struct ndpi_detection_module_struct
     if((memcmp(packet->payload, "RTSP/1.0 ", 9) == 0)
        || (strstr(buf, "rtsp://") != NULL)) {
       NDPI_LOG_DBG2(ndpi_struct, "found RTSP/1.0 \n");
-      if (dst != NULL) {
-	NDPI_LOG_DBG2(ndpi_struct, "found dst\n");
-	ndpi_packet_src_ip_get(packet, &dst->rtsp_ip_address);
-	dst->rtsp_timer = packet->current_time_ms;
-	dst->rtsp_ts_set = 1;
```
Comment (utoni) [inline src/lib/protocols/rtsp.c:91]: LoL, I did not know that OpenDPI used timestamps for flow correlation..
I am glad you removed that nonsense. :zany_face:

**Review by utoni — APPROVED** [review]

**Comment by vel21ripn** [comment]
Then you need to remove these fields from ndpi_id_struct

**Comment by IvanNardi** [comment]
> Then you need to remove these fields from ndpi_id_struct

I am not sure I understand your point. These fields *are* removed by this PR.
Am I missing something?

**Comment by lucaderi** [comment]
I think this is good work towards cleaning nDPI. How did you do this? By hand or do you have some code analysis tool for this?

**Comment by IvanNardi** [comment]
> I think this is good work towards cleaning nDPI. How did you do this? By hand or do you have some code analysis tool for this?

By hand, unfortunately... Focusing only on memory usage per flow, I had to check only a few structures, though.

**Pull request merged**